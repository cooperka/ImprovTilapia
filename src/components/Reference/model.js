import { initPersistedStore } from '../../storageUtils';

const STORAGE_KEY = 'ReferenceSettingsModel';

export class ReferenceSettingsModel {
  constructor() {
    initPersistedStore(this, STORAGE_KEY).catch((error) => {
      console.error('Failed to initPersistedStore:', error);
    });
  }
}
