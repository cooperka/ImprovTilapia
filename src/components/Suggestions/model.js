import { observable } from 'mobx';

import { initPersistedStore } from '../../storageUtils';

const STORAGE_KEY = 'SuggestionsSettingsModel';

export class SuggestionsSettingsModel {
  @observable
  shouldAutoSuggest = false;

  @observable
  autoSuggestIntervalSeconds = 10;

  constructor() {
    initPersistedStore(this, STORAGE_KEY).catch((error) => {
      console.error('Failed to initPersistedStore:', error);
    });
  }
}
