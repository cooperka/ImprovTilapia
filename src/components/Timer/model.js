import { observable } from 'mobx';

import { initPersistedStore } from '../../storageUtils';

const STORAGE_KEY = 'TimerSettingsModel';

export class TimerSettingsModel {
  @observable
  shouldIncreaseBrightness = true;

  constructor() {
    initPersistedStore(this, STORAGE_KEY).catch((error) => {
      console.error('Failed to initPersistedStore:', error);
    });
  }
}
