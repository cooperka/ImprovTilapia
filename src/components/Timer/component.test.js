import React from 'react';
import renderer from 'react-test-renderer';

import Component from './component';
import { TimerSettingsModel } from './model';

jest.useFakeTimers();
jest.mock('../FloatingNav/component', () => 'FloatingNav');

const mockTimerSettings = new TimerSettingsModel();

it('renders as expected', async () => {
  const tree = renderer.create(<Component timerSettings={mockTimerSettings} />);
  expect(tree.toJSON()).toMatchSnapshot();

  // TODO: Click buttons with Enzyme to actually test this.
  jest.advanceTimersByTime(1100);
  expect(tree.toJSON()).toMatchSnapshot();
});
