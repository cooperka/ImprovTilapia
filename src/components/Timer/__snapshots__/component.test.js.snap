// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<View
  className={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    className={Object {}}
  >
    0
  </Text>
  <Button
    icon={[Function]}
    mode="contained"
  />
</View>
`;

exports[`renders as expected 2`] = `
<View
  className={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    className={Object {}}
  >
    1
  </Text>
  <Button
    icon={[Function]}
    mode="contained"
  />
</View>
`;
